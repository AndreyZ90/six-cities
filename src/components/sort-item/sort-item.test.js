import React from 'react';
import TestRenderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import SortItem from '@/components/sort-item/sort-item';

const item = `Popular`;
const currentCity = `Amsterdam`;

describe(`SortItem component snapshot`, () => {
  test(`Should correctly render SortItem component (active)`, () => {
    const tree = TestRenderer.create(
        <Router>
          <SortItem item={item} isActive={true} currentCity={currentCity} onFlagChange={() => {}} />
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render SortItem component (not active)`, () => {
    const tree = TestRenderer.create(
        <Router>
          <SortItem item={item} isActive={false} currentCity={currentCity} onFlagChange={() => {}} />
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
