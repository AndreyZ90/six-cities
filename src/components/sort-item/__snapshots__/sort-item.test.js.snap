// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem component snapshot Should correctly render SortItem component (active) 1`] = `
<a
  href="/amsterdam?sort=Popular"
  onClick={[Function]}
>
  <li
    className="places__option places__option--active"
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
  </li>
</a>
`;

exports[`SortItem component snapshot Should correctly render SortItem component (not active) 1`] = `
<a
  href="/amsterdam?sort=Popular"
  onClick={[Function]}
>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
  </li>
</a>
`;
